### PR TITLE
gtkglext: strip pangox-compat dependency

### PR DESCRIPTION
**Pull request by vmasutin**
* strip pangox-compat dependency

#### Description

Stripping `pangox-compat` dependency (patch originates in debian, `pangox-compat` was abandoned long time ago and in unbuildable state, gtkglext is a sole dependant on it.
I was not able to test quartz variant (I don't have quartz installed)
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 13.1 22C65 arm64
Xcode 14.1 14B47b

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by reneeotten** [comment]
@vmasutin are you planning to work on following-up on @ryandesign his comment?

**Comment by ryandesign** [comment]
I hope you will check all of the ports that use gtkglext:

```
% port file all | sort -u | xargs grep -li ':gtkglext\b' | cut -d/ -f13
gcam
gtk2hs
gtkglextmm
gnubg
gliv
graphviz-devel
graphviz
openvrml
gfsview
ghemical
gwyddion
tempo
pcb
```

but I would be particularly interested in the results for ghemical since [ghemical is broken](https://trac.macports.org/ticket/67783) with the existing fix for pangox-compat.

**Comment by vmasutin** [comment]
> but I would be particularly interested in the results for ghemical since [ghemical is broken](https://trac.macports.org/ticket/67783) with the existing fix for pangox-compat.

It starts for me (I am getting lot of warnings though), is there any public file to load and check (I have never used it before)?

**Comment by vmasutin** [comment]
apps:
* `ghemical` starts, I was able to load test data
* `gcam` starts, looks functional
* `gnubg` starts, playable
* `gtk2hs` fails with ` checking for the GHC package "haskell98"... configure: error:`, I know nothing about haskell ecosystem, so can hardly help with this
* `gtkglextmm` library builds, did not find any projects that are using it (correct me if I am wrong)
* `gliv` starts, looks functional
* `graphviz` builds, looks functional (if you have specific test case, let me know)
* `openvrml` build fails (unclear why)
* `gfsview` starts, looks functional
* `gwyddion` starts, looks functional
* `tempo` builds, but I don't know how to test it (e.g. test file)
* `pcb` starts, looks functional

**Comment by pmetzger** [comment]
Can this be merged? It's been sitting in the queue quite a while.

**Comment by mascguy** [comment]
> Can this be merged? It's been sitting in the queue quite a while.

Forgot about this one, will take a look when I can.

Thanks for the heads-up Perry!

File: devel/gtkglext/Portfile [hunk 1]
```diff
@@ -24,12 +24,17 @@ depends_build       port:pkgconfig
 depends_lib         path:lib/pkgconfig/gtk+-2.0.pc:gtk2 \
                     port:libGLU \
                     port:mesa \
-                    port:pangox-compat \
                     port:xorg-libXmu
 
+use_autoreconf  yes
+autoreconf.args -fvi
```
[the remark this answers is not in the document]
Comment (mascguy) [reply devel/gtkglext/Portfile:30]: > If you are using autoreconf, you need to be patching configure.in and Makefile.am files. If you are not using autoreconf, you need to be patching configure and Makefile.in files. Your changes are currently inconsistent with regard to that. (You are using autoreconf, but in the quartz variant you are patching Makefile.in files.)

Yep, we'll need to conditionally use `autoreconf`, only for non-Quartz.

Currently working on testing/updating this to support Quartz as  well.

**Comment by mascguy** [comment]
Changing status to draft for the moment, while working through what's needed for Quartz.

File: devel/gtkglext/Portfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (ryandesign) [reply devel/gtkglext/Portfile:30]: Using autoreconf conditionally seems most likely to result in problems. I recommend using it either always or never and adjusting the patches to match whichever choice is made there.

[the remark this answers is not in the document]
Comment (mascguy) [reply devel/gtkglext/Portfile]: > Using autoreconf conditionally seems most likely to result in problems. I recommend using it either always or never and adjusting the patches to match whichever choice is made there.

I certainly agree in principle. Unfortunately the Quartz `configure` patch is huge, so it's going to take time to figure out.

Will try my best, though. So let's leave this as draft for now, until that's done.

**Comment by pmetzger** [comment]
Is there any progress on this? It's now approaching a year old.

**Comment by mascguy** [comment]
> Is there any progress on this? It's now approaching a year old.

Haven't forgotten about it, will add to my to-do list

**Comment by pmetzger** [comment]
It's been half a year now.

**Comment by pmetzger** [comment]
Closing this until such time as someone decides they want to finish it. Do recall that PRs can be reopened at will.

**Comment by mascguy** [comment]
> Closing this until such time as someone decides they want to finish it. Do recall that PRs can be reopened at will.

No worries, it's reasonable to close PRs that have been sitting idle for months.

Will get back to all of these at some point.

**Pull request closed without merging**